astropy/modeling/bounding_box.py

```diff
@@ -66,7 +66,7 @@ def _validate_shape(interval):
                 # np.shape does not work with lists of Quantities
                 if len(interval) == 1:
                     interval = interval[0]
-                shape = np.shape([b.to_value() for b in interval])
+                shape = np.shape([b.view(np.ndarray) for b in interval])
```

Comment:
As mentioned by @nden, it might make sense to instead fix ``SpectralCoord.to_value`` so that the unit is optional to match ``Quantity.to_value``. Could you update the PR to do this instead?

Reply:
This is more involved that one might think. [`SpectralCoord.to_value()`](https://github.com/astropy/astropy/blob/993cf238fe26fa543b31de0d147cd532d9d205e9/astropy/coordinates/spectral_quantity.py#L300-L301) just passes its arguments through to [`SpectralCoord.to()`](https://github.com/astropy/astropy/blob/993cf238fe26fa543b31de0d147cd532d9d205e9/astropy/coordinates/spectral_quantity.py#L180-L184), which does follow the interface laid out by [`Quantity.to()` ](https://github.com/astropy/astropy/blob/993cf238fe26fa543b31de0d147cd532d9d205e9/astropy/units/quantity.py#L816) in that it makes `unit` a required input.

Reply:
See commit 3167571.